src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let resultSum = 0;
+
+  const funAdder = (a) => {
+    if (a === undefined) {
+      const returnValue = resultSum;
+
+      resultSum = 0;
+
+      return returnValue;
+    } else {
+      resultSum += a;
```

Comment:
The addition operation is concise and correctly updates the `resultSum`. The logic here is sound and follows the task description.